playbooks/slave_security.yml

```diff
@@ -28,12 +34,27 @@
         port: 22
       when: ip_address_whitelist == ""
 
+    - name: Whitelist ports
+      ufw:
+        rule: allow
+        port: "{{ item }}"
+      with_items: "[ {{ ports_whitelist }} ]"
+
     - name: Enable firewall
       ufw:
         state: enabled
         policy: deny
 
-  roles:
-    - role: jnv.unattended-upgrades
+    - include_role:
+        name: willshersystems.sshd
+      vars:
+        sshd:
+          PasswordAuthentication: "no"
+          X11Forwarding: "no"
+          PrintLastLog: "no"
+          GatewayPorts: "yes"
+          Compression: "yes"
```

Comment:
If I'm reading this correctly, some instances like long running slaves will have the role included here, and then again in setup_jenkins_slave.yml (with a marginally different config).  Wonder if we can remove the inclusion of the role in setup_jenkins_slave.yml and then any sensitive instance just needs to ensure they're including slave_security.yml.

Reply:
I think we should register a card to figure out the best approach for the jenkins slave thing. Right now the two are separate and we'll need to flip the long running slave setup to use this mechanism IMO, and in so doing also figure things out like how we best handle their security, the ports, etc. I think this might be easy enough to handle with group_vars.

For this PR I''ll add another commit which ensures that the two role executions have the same results.

Reply:
OK ... in retrospect I don't think removing the ssh role from setup_jenkins_slave.yml would be a good idea since we don't want to allow password logins into our single use slaves since some of those create artifacts.    We also can't apply slave_security.yml to the single use slaves since the fw config, etc. will affect RPCO AIO deploys.

Reply:
Currently infrastructure instances are registered as Jenkins slaves, not for running normal jobs, but for targeting for maintenance etc. That doesn't seem to be the case here. Is the idea that we will use dynamic inventory instead of Jenkins for connecting to infrastructure nodes? If so why have a different mechanism for infrastructure nodes and short lived nodes?

Reply:
In this case we're both building and executing the builds from a remote 'deployment host' which is a jenkins slave. If the target is missing, it is replaced automatically. If were setting these up as slaves then it would be more complex to do that model I think. If we want to set these up as jenkins slaves too, I don't really understand what we gain from it.

Reply:
It's easier having a node as a slave, because it means you can execute ad-hoc commands on it.  However, for long-running infrastructure I think it makes more sense to have everything applied to a node captured in a playbook that is run end-to-end.